#!/usr/bin/env python

PACKAGE='final_aerial_project'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

onoff_enum          = gen.enum([gen.const("Off",                    str_t, "Off",        "Off"),
                                gen.const("On",                     str_t, "On",         "On") ],
                                "On or Off")


gen.add("gravity_compensation",  double_t,  0, "gravity_compensation", 15.0, 10.0, 20.0)

gen.add("x_kp",                  double_t,  0, "x_kp", 0.2, 0.0, 5.0)
gen.add("x_ki",                  double_t,  0, "x_ki", 0.0, 0.0, 5.0)
gen.add("x_kd",                  double_t,  0, "x_kd", 0.4, 0.0, 5.0)
gen.add("x_integral_limit",      double_t,  0, "x_integral_limit", 0.1, 0.0, 10.0)

gen.add("y_kp",                  double_t,  0, "y_kp", 0.2, 0.0, 5.0)
gen.add("y_ki",                  double_t,  0, "y_ki", 0.0, 0.0, 5.0)
gen.add("y_kd",                  double_t,  0, "y_kd", 0.4, 0.0, 5.0)
gen.add("y_integral_limit",      double_t,  0, "y_integral_limit", 0.1, 0.0, 10.0)

gen.add("z_kp",                  double_t,  0, "z_kp", 4.0, 0.0, 10.0)
gen.add("z_ki",                  double_t,  0, "z_ki", 2.2, 0.0, 10.0)
gen.add("z_kd",                  double_t,  0, "z_kd", 4.0, 0.0, 10.0)
gen.add("z_integral_limit",      double_t,  0, "z_integral_limit", 0.5, 0.0, 10.0)

gen.add("yaw_kp",                double_t,  0, "yaw_kp", 8.00, 0.0, 10.0)
gen.add("yaw_ki",                double_t,  0, "yaw_ki", 0.00, 0.0, 10.0)
gen.add("yaw_kd",                double_t,  0, "yaw_kd", 4.00, 0.0, 10.0)
gen.add("yaw_integral_limit",    double_t,  0, "yaw_integral_limit", 0.5, 0.0, 10.0)

gen.add("x_vel_limit",           double_t,  0, "x_vel_limit",   0.2, 0.0, 5.0)
gen.add("y_vel_limit",           double_t,  0, "y_vel_limit",   0.2, 0.0, 5.0)
gen.add("z_vel_limit",           double_t,  0, "z_vel_limit",   3.0, 0.0, 5.0)
gen.add("yaw_vel_limit",         double_t,  0, "yaw_vel_limit", 3.0, 0.0, 15.0)

exit(gen.generate(PACKAGE, "final_aerial_project", "Controller"))
